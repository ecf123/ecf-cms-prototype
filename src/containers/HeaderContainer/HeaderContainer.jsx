import "./HeaderContainer.scss";
import HeaderProfile from "../../components/HeaderProfile/HeaderProfile";
import PageTitle from "../../components/PageTitle/PageTitle";
const HeaderContainer = ({
  profileImg,
  name,
  role,
  toggle,
  titleSmall,
  titleLarge,
  classNameSmall,
  classNameLarge,
}) => {
  return (
    <div className="header-container">
      <div className="header-container__top">
        <PageTitle className={classNameSmall} title={titleSmall} />
        <HeaderProfile
          profileImg={profileImg}
          name={name}
          role={role}
          toggle={toggle}
        />
      </div>
      <PageTitle className={classNameLarge} title={titleLarge} />
    </div>
  );
};

export default HeaderContainer;
