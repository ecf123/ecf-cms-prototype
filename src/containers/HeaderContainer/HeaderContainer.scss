@use "../../styles/variables" as *;

.header-container {
  display: flex;
  flex-direction: column;
  padding: 0.625rem;
  background-color: $color-white;
  &__top {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }
}