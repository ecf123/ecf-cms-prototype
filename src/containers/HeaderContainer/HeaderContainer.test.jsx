import { customRender } from "../../utils/testUtils";
import HeaderContainer from "./HeaderContainer";

it("it should render the HeaderContainer", () => {
  const { container } = customRender(<HeaderContainer />);
  expect(container).toMatchSnapshot();
});

it("it should render the HeaderContainer with props", () => {
  const { container } = customRender(
    <HeaderContainer
      classNameSmall={"page-title page-title--small"}
      classNameLarge={"page-title page-title--large"}
      titleSmall={"Pathways"}
      titleLarge={"Financial Forest"}
      profileImg={"man"}
      name={"John Doe"}
      role={"Dev"}
      toggle={"toggle"}
    />
  );
  expect(container).toMatchSnapshot();
});
