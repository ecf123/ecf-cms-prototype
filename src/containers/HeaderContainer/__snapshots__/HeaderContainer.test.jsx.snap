// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the HeaderContainer 1`] = `
<div>
  <div
    class="header-container"
  >
    <div
      class="header-container__top"
    >
      <h1 />
      <div
        class="header-profile"
      >
        <img
          alt="user avatar"
          class="header-profile__image"
        />
        <div
          class="header-profile__text"
        >
          <h2
            class="header-profile__name"
          />
          <p
            class="header-profile__role"
          />
        </div>
        <button
          class="header-profile__arrow"
        >
          <img
            alt="toggle"
            src="upside-down-arrow-grey.svg"
          />
        </button>
      </div>
    </div>
    <h1 />
  </div>
</div>
`;

exports[`it should render the HeaderContainer with props 1`] = `
<div>
  <div
    class="header-container"
  >
    <div
      class="header-container__top"
    >
      <h1
        class="page-title page-title--small"
      >
        Pathways
      </h1>
      <div
        class="header-profile"
      >
        <img
          alt="user avatar"
          class="header-profile__image"
          src="man"
        />
        <div
          class="header-profile__text"
        >
          <h2
            class="header-profile__name"
          >
            John Doe
          </h2>
          <p
            class="header-profile__role"
          >
            Dev
          </p>
        </div>
        <button
          class="header-profile__arrow"
        >
          <img
            alt="toggle"
            src="upside-down-arrow-grey.svg"
          />
        </button>
      </div>
    </div>
    <h1
      class="page-title page-title--large"
    >
      Financial Forest
    </h1>
  </div>
</div>
`;
